[TensorExpr] Nuke TE shape inference.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* #65843
* __->__ #65554

We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

Commit:
[TensorExpr] Nuke TE shape inference.

We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -384,209 +384,12 @@ std::vector<ExprHandle> TensorExprKernel::sizesForValue(
+  GRAPH_DEBUG("Unknown sizes for the node: ", *v->node());
+  GRAPH_DEBUG("Full fusion group graph:\n", *v->node()->owningGraph());
+  std::string msg =
+      std::string("Unhandled node kind (in inferSizesForValue): ") +
```
Update this function name.

navahgar reviewed (commented):
Do we need more tests for both positive and negative cases here? Just to make sure that we don't throw exceptions in some corner cases.

bertmaher commented:
Broadcast + random almost certainly does not work correctly.  I think we do not fuse rand for this reason

Commit:
rebase on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

ZolotukhinM commented:
> Do we need more tests for both positive and negative cases here? Just to make sure that we don't throw exceptions in some corner cases.

Do you mean tests where shapes are not specified in the graph? I think this situation doesn't occur in the current setting (and that's the reason I'd like to kill this code - it's not actually used). In the near future I plan to add some passes that would verify that graph has all the info so that we can be sure TEK can handle it. Would it be ok with you if I add more tests along with that pass?

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

bertmaher commented:
I'm slightly frightened of this change since it reduces the number of if statements in front of a `throw` :-D (I kind of suspect @navahgar feels the same).  I'm wondering if we should backtest it on a bunch of production models...

ZolotukhinM commented:
I'm down to testing it just in case, I'll follow up with you guys to learn how exactly to do that :)

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).

Commit:
Update on "[TensorExpr] Nuke TE shape inference."


We're relying on JIT based shape inference and not using the TE
implementation.

Question to the audience: we set `hasBroadcasts_` in that function, but
this function was almost never invoked. Do we behave correctly in the
presence of rand-calls and broadcasts?

Differential Revision: [D31148925](https://our.internmc.facebook.com/intern/diff/D31148925)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148925).
